### PR TITLE
Adding Configuration.properties file and Property Reader

### LINKED ISSUE #28: Add Properties file and reader

### PR DESCRIPTION
**Pull request by dceaicovschi**

### CODE REVIEW

File: src/main/java/com/endava/soa_5/tools/PropertyReader.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    public PropertyReader() {
+        properties = new Properties();
+    }
+
+    public String getPropertyValue(String key) {
+        try {
+            inputStream = getClass().getClassLoader().getResourceAsStream(CONFIGURATION_FOLDER + "Configuration.properties");
+            properties.load(inputStream);
+
```
Comment (NadejdaClima) [inline src/main/java/com/endava/soa_5/tools/PropertyReader.java]: you can optimize your code by splitting the actions of reading file and getting the property into two separate methods,
then calling the reading method into the constructor(readProperties(path) for example).
That way, when running the program, the property file will only be read once and stored into an object, which we can use to extract values from (using the getProperty(String key) method)

**Review by NadejdaClima — APPROVED** [review]

File: src/main/java/com/endava/soa_5/tools/PropertyReader.java (hunk 1 above)
Comment (dceaicovschi) [reply src/main/java/com/endava/soa_5/tools/PropertyReader.java]: Implemented in another way.

**Review by NadejdaClima — APPROVED** [review]

**Review by danielachihai — APPROVED** [review]

**Review by ebernevec — APPROVED** [review]

File: src/main/java/com/endava/soa_5/tools/PropertyReader.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    private Properties properties;
+    private InputStream inputStream;
+    private String propertyValue;
+
+    public PropertyReader(String fileName) {
+        properties = new Properties();
+        properties = loadConfiguration(fileName);
+    }
+
+    public Properties loadConfiguration(String fileName) {
```
Comment (MarcelMorozniuc) [inline src/main/java/com/endava/soa_5/tools/PropertyReader.java]: can be private

File: src/main/java/com/endava/soa_5/tools/PropertyReader.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            ioException.printStackTrace();
+        }
+        return properties;
+    }
+
+    public String getPropertyValue(String key) {
+        try {
+            String controlValue = properties.getProperty(key);
+            if (controlValue != null) {
+                propertyValue = controlValue;
```
Comment (MarcelMorozniuc) [inline src/main/java/com/endava/soa_5/tools/PropertyReader.java]: please consider that instead creating new variable, can be simply returned controlValue in if(), else can be removed, and in the end returned null (so in case that something was happening it will return null)

File: src/test/java/com/endava/soa_5/tools/PropertyReaderTest.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+class PropertyReaderTest {
+    PropertyReader propertyReader;
+
+    @BeforeEach
+    void setUp() {
+        propertyReader = new PropertyReader("Configuration.properties");
+    }
+
+    @Test
```
Comment (MarcelMorozniuc) [inline src/test/java/com/endava/soa_5/tools/PropertyReaderTest.java:18]: If decided cover functionality with tests, a negative one can be added

File: src/main/java/com/endava/soa_5/tools/PropertyReader.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                inputStream.close();
```
Comment (MarcelMorozniuc) [inline src/main/java/com/endava/soa_5/tools/PropertyReader.java:35]: can be moved to finally

File: src/main/java/com/endava/soa_5/tools/PropertyReader.java (hunk 2 above)
Comment (dceaicovschi) [reply src/main/java/com/endava/soa_5/tools/PropertyReader.java]: Done!

File: src/main/java/com/endava/soa_5/tools/PropertyReader.java (hunk 5 above)
Comment (dceaicovschi) [reply src/main/java/com/endava/soa_5/tools/PropertyReader.java:35]: Done!

File: src/test/java/com/endava/soa_5/tools/PropertyReaderTest.java (hunk 4 above)
Comment (dceaicovschi) [reply src/test/java/com/endava/soa_5/tools/PropertyReaderTest.java:18]: Done!

File: src/main/java/com/endava/soa_5/tools/PropertyReader.java (hunk 3 above)
Comment (dceaicovschi) [reply src/main/java/com/endava/soa_5/tools/PropertyReader.java]: Done!

**Review by MarcelMorozniuc — APPROVED** [review]
